pandas/io/excel.py

```diff
@@ -373,6 +388,12 @@ def _parse_cell(cell_contents, cell_typ):
                 val = int(cell_contents)
                 if val == cell_contents:
                     cell_contents = val
+            elif true_values and cell_contents in true_values:
+                # GH13347
```

Comment:
Just so I'm clear you are asking to change it from:
`elif true_values and cell_contents in true_values:`
to
`elif true_values and cell_contents in set(true_values):`